File: NC_primeiro_carro/_primeiro_carro.py
meses, Y = map(float,input().split())
X = [float(input()) for _ in range(int(meses))]

caixa = 0.00
mes = 1
comprou = False

for valor in X:
    caixa += valor
    if caixa >=Y:
        comprou = True
    else:
        mes+=1

if comprou:
    print(f'{mes} {caixa-Y:.2f}')
else:
    print(f'0 {caixa:.2f}')

File: NC_primeiro_carro/_primeiro_carro.cpp
#include <iostream>
using namespace std;

int main(){
    int Meses, Preco_do_carro, Mes;
    float S, caixa,res;
    bool comprou = false;
    cin >> Meses >> Preco_do_carro;
    int X[Meses];
    for(int i=0;i<Meses;i++){
        cin >> S;
        X[i] = S;
    }

    for(int j : X){
        caixa += X[j];
        cout << "O valor da caixa:" << caixa << endl;
        if (caixa>= Preco_do_carro){
            comprou = true;
        }
        else{
            Mes+=1;
        }
    }
    if (comprou){
        res = caixa-Preco_do_carro;
        cout << Mes << (res);
    }
    else{
        cout << "0" << caixa;
    }

}